docs/source/1.0/spec/core/behavior-traits.rst

```diff
@@ -655,3 +655,223 @@ See
+            "shapes": {
+                "smithy.example#Example": {
+                    "type": "service",
+                    "version": "2019-06-27",
+                },
+                "smithy.example#PutSomething": {
+                    "type": "operation",
+                    "input": {
+                        "target": "smithy.example#PutSomethingInput"
+                    },
+                    "output": {
+                        "target": "smithy.example#PutSomethingOutput"
+                    },
+                    "traits": {
+                        "smithy.api#httpChecksum": {
+                            "request": [
+                                { "algorithm": "sha256", "in": "header", "name": "x-checksum-sha256"},
+                                { "algorithm": "crc32", "in": "header", "name": "x-checksum-crc32"}
+                            ],
+                            "response": [
+                                { "algorithm": "sha256", "in": "header", "name": "x-checksum-sha256"},
+                                { "algorithm": "crc32", "in": "header", "name": "x-checksum-crc32"}
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+
+.. _checksum-property:
+
+HttpChecksumProperty structure
+================================
+
+``HttpChecksumProperty`` defines checksum validation behavior using the
+following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+
+    * - algorithm
+      - ``string``
+      - **Required**. The algorithm string represents a checksum algorithm
+        supported for the HTTP request or response. A valid algorithm
+        should follow ABNF:
+
+        .. code-block:: abnf
+
+            algorithms = 1*(lower-alpha / DIGIT)
+            lower-alpha = %x61-7A
+
+        That is, it should be comprised only of digits and lowercase letters.
+
+    * - in
+      - ``string``
+      - **Required**. The location where checksum can be supplied. Valid values
+        are:
+
+        * ``header`` - Indicates the checksum should be placed in an HTTP header.
+        * ``trailer`` - Indicates the checksum should be placed in the `chunked trailer part`_
+          of the body.
+
+        An algorithm may be supported at multiple locations. The first
+        supported ``in`` member value within the list of HttpChecksum property
+        takes precedence and SHOULD be used. See :ref:`here for more behavior
+        details <httpChecksum_trait_behavior>`.
+
+    * - name
+      - ``string``
+      - **Required**. The name used as a header or trailer name for a checksum
+        type. Recommended ABNF for name is as follows:
+
+        .. code-block:: abnf
+
+            name = lower-alpha *(["-"](lower-alpha / DIGIT))
+            lower-alpha = %x61-7A
+
+        That is, it should start with a lowercase letter followed by any
```

Comment:
Can you give some context on why it's much more restrictive than the field-name production in [RFC 7230, section 3](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2)? I'm assuming it's because headers are case-insensitive, and we want to normalize it here?

Reply:
Correct, this is to normalize the name and stay consistent.